### PR TITLE
Add tests for enum usage as transaction parameters

### LINKED ISSUE #3484: Support for Enum Types as Direct Transaction Parameters
### Issue to be solved

**Issue to be Solved**

Currently, enums cannot be used directly as transaction parameters in the Cadence language. This limitation requires developers to use integer values to represent enum cases, which reduces code readability and increases the risk of errors due to incorrect mappings between integers and their corresponding enum cases. This issue affects the security and usability of smart contracts, as users must be aware of the specific integer values representing each enum case, leading to potential mistakes and confusion.

**In-scope:**

- Allowing enum types to be directly used as transaction parameters.

- Improving the security and usability of smart contracts by enabling intuitive use of enums.

**Out-of-scope:**

- Changing existing enum functionalities or how enums are defined within contracts.

- Modifying how integer values are currently used in other parts of the language.

### Suggested Solution

**Details of the Technical Implementation:**

- Update the Cadence language to support enum types as valid transaction parameters.

- Allow developers to define enums within contracts and use them directly in transaction arguments.

- Ensure that the Cadence runtime can correctly interpret and validate enum values during transaction execution.

**Tradeoffs Made in Design Decisions:**

- Pro: Enhanced readability and reduced risk of errors by using enums directly.

- Pro: Improved developer experience and code maintainability.

- Con: Requires updates to the Cadence language and runtime, which may involve significant development resources and testing.

**Caveats and Considerations for the Future:**

**Ensure** backward compatibility with existing contracts that use integers to represent enums.
**Provide** thorough documentation and examples to help developers transition to using enums as transaction parameters.
**Consider** potential performance implications of supporting enums in transaction parameters and optimize as necessary.

**Example of Preferred Solution:**

Currently, a transaction script might require the user to input an integer value to represent an enum case, such as:

```cadence
import ExampleContract from 0x01

transaction(enumRawValue: UInt8) {
    let adminRef: &ExampleContract.Admin

    prepare(signer: AuthAccount) {
        self.adminRef = signer.borrow<&ExampleContract.Admin>(from: /storage/Admin)
            ?? panic("Could not borrow a reference to the Admin resource")
    }

    execute {
        let enumValue: ExampleContract.EnumType = ExampleContract.EnumType(rawValue: enumRawValue) ?? panic("Invalid enum value")
        self.adminRef.updateEnumValue(value: enumValue)
    }
}
```

In this script, the user must provide an integer representing the enum value. With the proposed enhancement, the script could be simplified to:

```cadence
import ExampleContract from 0x01

transaction(enumValue: ExampleContract.EnumType) {
    let adminRef: &ExampleContract.Admin

    prepare(signer: AuthAccount) {
        self.adminRef = signer.borrow<&ExampleContract.Admin>(from: /storage/Admin)
            ?? panic("Could not borrow a reference to the Admin resource")
    }

    execute {
        self.adminRef.updateEnumValue(value: enumValue)
    }
}
```

### PR DESCRIPTION
**Pull request by RZhang05**
Closes #3484 

## Description

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by SupunS — COMMENTED** [review]
Thank you @RZhang05 for adding tests!

It seems this feature is already supported, so nothing really to add. Could you maybe remove the draft status from the PR, so we can get these tests in.

File: runtime/program_params_validation_test.go [hunk 1]
```diff
@@ -1391,4 +1391,86 @@ func TestRuntimeTransactionParameterTypeValidation(t *testing.T) {
...
+				  case B
+			  }
+		   }
+		`),
+		}
+
+		script := `
+	  import C from 0x1
+
+	  transaction(arg: C.Alpha) {}
```
Comment (turbolent) [inline runtime/program_params_validation_test.go:1469]: Maybe extend the body here a bit and use `arg`, e.g. by comparing it, logging it, etc.

**Review by turbolent — APPROVED** [review]
Nice! Great that this was already working.

File: runtime/program_params_validation_test.go (hunk 1 above)
Comment (RZhang05) [reply runtime/program_params_validation_test.go]: Done.

File: runtime/program_params_validation_test.go [hunk 2]
```diff
@@ -1391,4 +1391,95 @@ func TestRuntimeTransactionParameterTypeValidation(t *testing.T) {
...
+			}: []byte(`
+		   access(all) contract C {
+			   access(all) 
+			   enum Alpha: Int {
+				  access(all)
+				  case A
+
+				  access(all)
+				  case B
+			  }
```
Comment (SupunS) [inline runtime/program_params_validation_test.go:1430]: Just a nit-pick: seems there are mix indentations of whitespaces and tabs. Maybe use only one kind (e.g: whitespaces only) for indentations.

**Review by SupunS — APPROVED** [review]
👍

File: runtime/program_params_validation_test.go (hunk 2 above)
Comment (RZhang05) [reply runtime/program_params_validation_test.go]: Switched to tabs for consistency.

**Review by SupunS — APPROVED** [review]

**Pull request merged**